src/components/Fieldset/DynamicFieldset/ControlledDynamicFieldset.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import clsx from 'clsx';
+import '../Fieldset.scss';
+import type {ControlledDynamicFieldsetProps} from './DynamicFieldset.types';
+import {Switch, Typography} from '~/components';
+
+export const ControlledDynamicFieldset = React.forwardRef<HTMLDivElement, ControlledDynamicFieldsetProps>(({
+    id,
+    label,
+    helper,
+    children,
+    className,
+    buttons,
+    isOpen = false,
+    onChange,
+    ...props
+}, ref) => {
+    return (
+        <div
+            ref={ref}
+            id={id}
+            className={clsx(
+                'moonstone-dynamic-fieldset',
+                isOpen && 'moonstone-dynamic-fieldset_open',
+                'flexCol_nowrap',
+                className
+            )}
+            {...props}
+        >
+            <div className={clsx('flexRow_nowrap', 'flexFluid', 'alignCenter')}>
+                <div className="flexRow_nowrap flexFluid alignCenter">
+                    <Typography isNowrap component="label" variant="heading" weight="bold">{label}</Typography>
```

Comment:
You can use `legend` tag instead to set the title of a `fieldset` ([MDN doc](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/legend))